docs/design-ux/sharing.md

```diff
@@ -0,0 +1,120 @@
+# Sharing
+
+The sharing UX needs to balance simplicity with flexibility. Users need to be able to clearly tell who can access which
+documents and easily modify that access. They also need to be able to accomplish a wide variety of sharing-related
+tasks, including:
+
+* sharing individual documents or whole folders
+* sharing with readonly vs read/write access
+* managing who is billed for shared documents
+* linking to shared documents from shared documents and from unshared documents
+
+We evaluate 3 design alternatives according to their suitability for individuals, teams, and enterprises. Individuals
+are users who store thoughts in lockbook and occassionally want to share a file with another user. Teams are small groups
+of users who generally want someplace to store personal things and someplace to store team things, which are shared with
+the whole team. Enterprises are large organizations that want to control how things are shared in sophisticated ways, such
+as having teams with subteams that have at least as much access. The designs we consider are:
+
+1. Root sharing
+2. Folder sharing
+3. Policy Sharing
+
+## 1. Root Sharing
+
+### UX
```

Comment:
Proposition for one-off sharing UX: make a new vault for the group of people you want to share with, move the document to that vault, and replace it with a symlink to that document. Opening a symlink in the UI is the same as opening the document itself, so laypeople can't tell the difference. Symlinks are by UUID, are immutable, and can't point to other symlinks. Maybe shared vaults can't contain symlinks as an additional simplification?

Vaults like this would just sort of accumulate over time, like group chats. Instead of making formal teams like 1password or github, the vault could just be defined in terms of all the sharees, like how if you send a group message to the same 3 people as a previous message, they show up in the same thread. You can manage group members, name the group, etc like in messengers